code/modules/mob/mob.dm

```diff
@@ -610,11 +610,19 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
 		return 0
 
 	changeNext_move(CLICK_CD_POINT)
-	var/obj/P = new /obj/effect/temp_visual/point(tile)
-	P.invisibility = invisibility
-	P.pixel_x = A.pixel_x
-	P.pixel_y = A.pixel_y
-	return 1
+
+	var/image/I = image('icons/mob/screen_gen.dmi', tile, "arrow", POINT_LAYER)
+	I.pixel_x = A.pixel_x
+	I.pixel_y = A.pixel_y
+	I.invisibility = invisibility
+	I.plane = GAME_PLANE
+
+	var/list/viewers = list()
+	for(var/mob/M in range(src))
+		if(M.client && (src in view(M)))
```

Comment:
Don't you need to specify the tile distance for the `range` and `view` procs? I admit I don't actually know what happens if you use them like this.

Reply:
The documentation is a bit odd. But I believe using it like this will use the view range of the given object. http://www.byond.com/docs/ref/#/proc/view has an example of this.

Reply:
Ah alright, and so `range` apparently does the same thing but visibility is ignored according to the docs. I guess we can assume this is the full screen's worth of range.

Reply:
Aye that is what I assume atleast. I'll test it a bit more around the edge cases to see if this is exactly what it does

Reply:
Tested it a bit more. It indeed only works if you can see the one pointing and can see the thing being pointed at.